### PR TITLE
Add File\LoaderInterface with basic curl support

### PR DESCRIPTION
**Pull request by mlocati**
What about adding a simple loader that can fetch remote files with curl?
This would solve #415, and fix the problem that GraphicsMagick can't load images from https.

### CODE REVIEW

File: lib/Imagine/File/Loader.php [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    {
+        if (is_object($path) && !method_exists($path, '__toString')) {
+            throw new InvalidArgumentException(sprintf('$path is an object of file %s which does not implement the __toString method', get_class($path)));
+        }
+
+        $this->path = (string) $path;
+        if ($this->path === '') {
+            throw new InvalidArgumentException('$path is empty');
+        }
+        $this->isUrl = filter_var($this->path, FILTER_VALIDATE_URL);
```
Comment (dmarkowicz) [inline lib/Imagine/File/Loader.php]: Should be casted to bool, now url is assigned.

File: tests/Imagine/Test/Image/AbstractImagineTest.php [hunk 2]
```diff
@@ -116,7 +116,7 @@ public function testShouldCreateImageFromString()
 
     public function testShouldCreateImageFromStreamWithMetadata()
     {
-        $source = 'http://imagine.readthedocs.org/en/latest/_static/exit-90-test.jpg';
+        $source = 'https://imagine.readthedocs.org/en/latest/_static/exit-90-test.jpg';
```
Comment (dmarkowicz) [inline tests/Imagine/Test/Image/AbstractImagineTest.php]: Most likely you want to test it in testShouldOpenAnHttpImage() and change HTTP_IMAGE to https. Here image is opened by `fopen`.

Comment (mlocati) [reply tests/Imagine/Test/Image/AbstractImagineTest.php]: I'll revert this edit. About changing the HTTP_IMAGE constant: it should work even if we leave it with the HTTP protocol (the standard CURL adapter is configured to follow redirects).

**Comment by mlocati** [comment]
I'm going to merge this because it's required to let tests pass (it's blocking other pull requests).

**Pull request merged**